### ISSUE TITLE
dbg_macro lint not working

### ISSUE THREAD

**Issue by vmalloc** [rust-lang/rust-clippy#5963]
I tried this code:

```rust
fn main() {
    dbg!(1);
}
```

I expected to see this happen: given the dbg_macro lint, I expected clippy to warn me about the usage of `dbg!`

Instead, this happened:
```
Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (d3fb005a3 2020-07-31)
- `rustc -Vv`:
  ```rustc 1.45.2 (d3fb005a3 2020-07-31)
       binary: rustc
       commit-hash: d3fb005a39e62501b8b0b356166e515ae24e2e54
       commit-date: 2020-07-31
       host: x86_64-apple-darwin
       release: 1.45.2
       LLVM version: 10.0```

**Comment by ebroto** [comment]
The `dbg_macro` is in the `restriction` group, so it's not enabled by default.

This would work:
```rust
#![deny(clippy::dbg_macro)]

fn main() {
    dbg!(1);
}
```
([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4392f2626d3d55bb2a1cb63c14b320b1))

See [here](https://doc.rust-lang.org/reference/attributes/diagnostics.html) for more info.

**Comment by vmalloc** [comment]
@ebroto ah I see. Missed that one.
Thanks and sorry for the noise!
